sdks/go/absurd/absurd.go

```diff
@@ -0,0 +1,644 @@
+		return err
+	}
+
+	c := defaultEmitEventConfig
+	for _, opt := range opts {
+		if err := opt(&c); err != nil {
+			return err
+		}
+	}
+
+	// Parse correct optional values.
+	queue := cmp.Or(c.Queue, a.base.Queue)
+	if queue == "" {
+		// Panic on invariant.
+		panic("absurd: default base queue not set")
+	}
+
+	// Emit event to postgres.
+	conn, err := a.base.connect(ctx)
+	if err != nil {
+		return err
+	}
+
+	rows, err := conn.Query(ctx, `SELECT absurd.emit_event($1, $2, $3)`, queue, eventName, payload)
+	if err != nil {
+		return err
+	}
+	rows.Close()
+
+	return rows.Err()
+}
+
+// -------------------------
+// Worker related operations
+// -------------------------
+
+// ClaimTasks claims up to batchSize tasks from the queue for processing.
+func ClaimTasks(
+	ctx context.Context,
+	a *Absurd,
+	batchSize int,
+	claimTimeout time.Duration,
+	workerID string,
+) ([]ClaimedTask, error) {
+	if batchSize < 1 {
+		return nil, ErrInvalidBatchSize
+	}
+	claimTimeout = claimTimeout.Round(time.Second)
+	if claimTimeout < time.Second {
+		return nil, ErrInvalidClaimTimeout
+	}
+
+	queue := a.base.Queue
+	if queue == "" {
+		panic("absurd: default base queue not set")
+	}
+
+	conn, err := a.base.connect(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := conn.Query(
+		ctx,
+		`SELECT run_id, task_id, attempt, task_name, params, retry_strategy, max_attempts,
+		        headers, wake_event, event_payload
+		 FROM absurd.claim_task($1, $2, $3, $4)`,
+		queue,
+		workerID,
+		int(claimTimeout.Seconds()),
+		batchSize,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var tasks []ClaimedTask
+	for rows.Next() {
+		var task ClaimedTask
+		if err := rows.Scan(
+			&task.RunID,
+			&task.TaskID,
+			&task.Attempt,
+			&task.TaskName,
+			&task.Params,
+			&task.RetryStrategy,
+			&task.MaxAttempts,
+			&task.Headers,
+			&task.WakeEvent,
+			&task.EventPayload,
+		); err != nil {
+			return nil, err
+		}
+		tasks = append(tasks, task)
+	}
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return tasks, nil
+}
+
+// NOTE(hohmannr): This is the equivalent to `startWorker`. I renamed it, but we can always change
+// it back.
+func Run(ctx context.Context, a *Absurd, opts ...RunOption) error {
+	var runningErrs []error
+
+	c := defaultRunConfig
+	for _, opt := range opts {
+		if err := opt(&c); err != nil {
+			return err
+		}
+	}
+
+	// Initialize runner state and error if already running.
+	a.runMu.Lock()
+	if a.runCancel != nil {
+		a.runMu.Unlock()
+		return ErrWorkerAlreadyRunning
+	}
+
+	// Create context that can be cancelled via [Close].
+	ctx, cancel := context.WithCancel(ctx)
+	a.runCancel = cancel
+	defer func() {
+		a.runMu.Lock()
+		a.runCancel = nil
+		a.runMu.Unlock()
+	}()
+
+	// Main polling loop.
+polling:
+	for {
+		// Block until we waited for poll interval or context was cancelled.
+		select {
+		case <-ctx.Done():
+			break polling
+		case <-time.After(c.PollInterval):
+		}
```

Comment:
it'd be nice to integrate with the listen/notify thing that postgresql has. It was mentioned in this issue https://github.com/earendil-works/absurd/issues/33 as a potential notification mechanism